prov/efa/src/rxr/rxr_read.c

```diff
@@ -79,15 +79,46 @@ int rxr_locate_rma_iov_pos(struct fi_rma_iov *rma_iov, int rma_iov_count, size_t
 	return -1;
 }
 
+ssize_t rxr_read_mr_reg(struct rxr_ep *ep, struct rxr_read_entry *read_entry)
+{
+	size_t i;
+	int err;
+
+	for (i = 0; i < read_entry->iov_count; ++i) {
+		if (read_entry->mr_desc[i] || read_entry->mr[i]) {
+			continue;
+		}
+
+		err = fi_mr_reg(rxr_ep_domain(ep)->rdm_domain,
+				read_entry->iov[i].iov_base, read_entry->iov[i].iov_len,
+				FI_RECV, 0, 0, 0, &read_entry->mr[i], NULL);
+
+		if (err) {
+			/* In case error happened, we did not close registered MR here, because:
+			 * if error is -FI_ENOMEM, we return -FI_EAGAIN, and the registration will
+			 * be retried later.
+			 * for other error, the caller will release read_entry, thus close MR
+			 */
+			FI_WARN(&rxr_prov, FI_LOG_MR, "Unable to register MR buf for read!\n");
+			if (err == -FI_ENOMEM)
+				err = -FI_EAGAIN;
+			return err;
+		}
+
+		read_entry->mr_desc[i] = fi_mr_desc(read_entry->mr[i]);
+	}
+
+	return 0;
+}
+
 struct rxr_read_entry *rxr_read_alloc_entry(struct rxr_ep *ep, int entry_type, void *x_entry,
```

Comment:
If you update a function, please take time to add documentation comments.

Reply:
Thanks! added in the new revision.